Avatar Token HouseKeeping (#16662)
MetaMask/metamask-extension: ui/components/component-library/avatar-token/README.mdx

  <Story id="ui-components-component-library-avatar-token-avatar-token-stories-js--size" />
</Canvas>

```jsx
import { AvatarToken } from '../../component-library';

  <AvatarToken size={SIZES.XS} />
  <AvatarToken size={SIZES.SM} />
  <AvatarToken size={SIZES.MD} />
  <AvatarToken size={SIZES.LG} />
  <AvatarToken size={SIZES.XL} />
```

### Name

Use the `name` prop to set the initial letter of the `AvatarToken`. This will be used as the fallback display if no image url is passed to the `src` prop.

<Canvas>
  <Story id="ui-components-component-library-avatar-token-avatar-token-stories-js--name" />
</Canvas>

```jsx
import { AvatarToken } from '../../component-library';

  <AvatarToken name="A" />
  <AvatarToken name="G" />
```

### Src

Use the `src` prop to set the image to be rendered of the `AvatarToken`.

<Canvas>
  <Story id="ui-components-component-library-avatar-token-avatar-token-stories-js--src" />
</Canvas>

```jsx
import { AvatarToken } from '../../component-library';

>   <AvatarToken src="./AST.png" />
>   <AvatarToken src="./BAT_icon.svg" />

Reviewer: nit: remove spaces/tabs

```suggestion
<AvatarToken src="./AST.png" />
<AvatarToken src="./BAT_icon.svg" />
```

```

### Show Halo

> If we want to display the component with halo effect. Only works if an image url is supplied to `src`

Reviewer: I think in the description of the prop we want to specifically mention it like in `src` and `size`

```suggestion
Use the `showHalo` prop to display the component with halo effect. Only works if an image url is supplied to `src`
```

<Canvas>
  <Story id="ui-components-component-library-avatar-token-avatar-token-stories-js--show-halo" />
</Canvas>

```jsx
import { AvatarToken } from '../../component-library';

>   <AvatarToken src="./AST.png" showHalo/>
>   <AvatarToken src="./BAT_icon.svg" showHalo/>

Reviewer: nit: remove tabs/spaces

```suggestion
<AvatarToken src="./AST.png" showHalo/>
<AvatarToken src="./BAT_icon.svg" showHalo/>
```

```

### Color, Background Color And Border Color

Use the `color`, `backgroundColor` and `borderColor` props to set the text color, background-color and border-color of the `AvatarToken`.

<Canvas>
  <Story id="ui-components-component-library-avatar-token-avatar-token-stories-js--color-background-color-and-border-color" />
</Canvas>

```jsx
import { COLORS } from '../../../helpers/constants/design-system';
import { AvatarToken } from '../../component-library';

<AvatarToken
  backgroundColor={COLORS.GOERLI}
  borderColor={COLORS.GOERLI}
  name="G"
  color={COLORS.PRIMARY_INVERSE}

>   >

Reviewer: nit: formatting
```suggestion
>
```

</Canvas>

```jsx
import { AvatarToken } from '../../component-library';

<AvatarToken src="./images/eth_logo.svg" />
<AvatarToken src="./images/arbitrum.svg" />
<AvatarToken src="./images/bnb.png" />
<AvatarToken src="https://static.metaswap.codefi.network/api/v1/tokenIcons/1/0x6b175474e89094c44da98b954eedeac495271d0f.png" />
<AvatarToken src="https://static.metaswap.codefi.network/api/v1/tokenIcons/1/0x0d8775f648430679a709e98d2b0cb6250d2887ef.png" />
<AvatarToken src="https://static.metaswap.codefi.network/api/v1/tokenIcons/1/0x7d1afa7b718fb893db30a3abc0cfc608aacfebb0.png" />
<AvatarToken src="https://i.seadn.io/gae/lSm8ChaI-3RqC9MTpi0j3KBXdfdPd57PN5UeQLY49JA3twy9wSt2dpaa22sSc6oyiXi2OEUR6GeFX8jwkZHEMADu6_Bd4EwTJ-rg?w=500&auto=format" />
```

### Show Halo

Use the `showHalo` prop to display the component with halo effect. Only works if an image url is supplied to `src`

<Canvas>
  <Story id="ui-components-component-library-avatar-token-avatar-token-stories-js--show-halo" />
</Canvas>

```jsx
import { AvatarToken } from '../../component-library';

<AvatarToken src="./AST.png" showHalo/>
<AvatarToken src="./BAT_icon.svg" showHalo/>

> ```

Reviewer: Update code example to reflect story. I was going to suggest using some other images here but once the eth image has been updated to the new one it should be ok

![image](https://user-images.githubusercontent.com/8112138/204008698-7b7fabcc-da56-46d5-bd27-7699711d6108.jpeg)

### Color, Background Color And Border Color

Use the `color`, `backgroundColor` and `borderColor` props to set the text color, background-color and border-color of the `AvatarToken`.

<Canvas>
  <Story id="ui-components-component-library-avatar-token-avatar-token-stories-js--color-background-color-and-border-color" />
</Canvas>

```jsx
import { COLORS } from '../../../helpers/constants/design-system';
import { AvatarToken } from '../../component-library';

<AvatarToken
  backgroundColor={COLORS.GOERLI}
  borderColor={COLORS.GOERLI}
  name="G"
  color={COLORS.PRIMARY_INVERSE}
>
  G

> </AvatarBase>

Author: Done

<AvatarToken src="https://i.seadn.io/gae/lSm8ChaI-3RqC9MTpi0j3KBXdfdPd57PN5UeQLY49JA3twy9wSt2dpaa22sSc6oyiXi2OEUR6GeFX8jwkZHEMADu6_Bd4EwTJ-rg?w=500&auto=format" />
```

### Show Halo

Use the `showHalo` prop to display the component with halo effect. Only works if an image url is supplied to `src`

<Canvas>
  <Story id="ui-components-component-library-avatar-token-avatar-token-stories-js--show-halo" />
</Canvas>

```jsx
import { AvatarToken } from '../../component-library';

<Box display={DISPLAY.FLEX} gap={1}>
<AvatarToken src="./images/eth_logo.svg" />
<AvatarToken src="./images/arbitrum.svg" />
<AvatarToken src="./images/bnb.png" />
<AvatarToken src="https://static.metaswap.codefi.network/api/v1/tokenIcons/1/0x6b175474e89094c44da98b954eedeac495271d0f.png" />
<AvatarToken src="https://static.metaswap.codefi.network/api/v1/tokenIcons/1/0x0d8775f648430679a709e98d2b0cb6250d2887ef.png" />
<AvatarToken src="https://static.metaswap.codefi.network/api/v1/tokenIcons/1/0x7d1afa7b718fb893db30a3abc0cfc608aacfebb0.png" />
<AvatarToken src="https://i.seadn.io/gae/lSm8ChaI-3RqC9MTpi0j3KBXdfdPd57PN5UeQLY49JA3twy9wSt2dpaa22sSc6oyiXi2OEUR6GeFX8jwkZHEMADu6_Bd4EwTJ-rg?w=500&auto=format" />

> </Box>

Author: Good Catch
